hive-metastore/src/main/java/org/apache/iceberg/hive/HiveViewOperations.java

```diff
@@ -0,0 +1,346 @@
+    return super.current();
+  }
+
+  @Override
+  public void doRefresh() {
+    String metadataLocation = null;
+    try {
+      Table table = metaClients.run(client -> client.getTable(database, tableName));
+      HiveCatalogUtil.validateTableIsIcebergView(table, fullName);
+      metadataLocation =
+          table.getParameters().get(BaseMetastoreTableOperations.METADATA_LOCATION_PROP);
+    } catch (NoSuchObjectException e) {
+      if (currentMetadataLocation() != null) {
+        throw new NoSuchViewException("View does not exist: %s.%s", database, tableName);
+      }
+    } catch (TException e) {
+      String errMsg =
+          String.format("Failed to get view info from metastore %s.%s", database, tableName);
+      throw new RuntimeException(errMsg, e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted during refresh", e);
+    }
+    refreshFromMetadataLocation(metadataLocation);
+  }
+
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  @Override
+  public void doCommit(ViewMetadata base, ViewMetadata metadata) {
+    boolean newTable = base == null;
+    String newMetadataLocation = writeNewMetadataIfRequired(metadata);
+    boolean keepHiveStats = conf.getBoolean(ConfigProperties.KEEP_HIVE_STATS, false);
+
+    boolean updateHiveTable = false;
+
+    try {
+
+      Table tbl = loadHmsTable();
+
+      if (tbl != null) {
+        // If we try to create the view but the metadata location is already set, then we had a
+        // concurrent commit
+        if (newTable
+            && tbl.getParameters().get(BaseMetastoreTableOperations.METADATA_LOCATION_PROP)
+                != null) {
+          throw new AlreadyExistsException("View already exists: %s.%s", database, tableName);
+        }
+
+        updateHiveTable = true;
+        LOG.debug("Committing existing view: {}", fullName);
+      } else {
+        tbl = newHmsTable(metadata);
+        LOG.debug("Committing new view: {}", fullName);
+      }
+
+      tbl.setSd(storageDescriptor(metadata)); // set to pickup any schema changes
+
+      String metadataLocation =
+          tbl.getParameters().get(BaseMetastoreTableOperations.METADATA_LOCATION_PROP);
+      String baseMetadataLocation = base != null ? base.metadataFileLocation() : null;
+      if (!Objects.equals(baseMetadataLocation, metadataLocation)) {
+        throw new CommitFailedException(
+            "Base metadata location '%s' is not same as the current table metadata location '%s' for %s.%s",
+            baseMetadataLocation, metadataLocation, database, tableName);
+      }
+
+      setHmsTableParameters(newMetadataLocation, tbl, metadata);
+
+      if (!keepHiveStats) {
+        tbl.getParameters().remove(StatsSetupConst.COLUMN_STATS_ACCURATE);
+      }
+
+      try {
+        persistTable(tbl, updateHiveTable, baseMetadataLocation);
+
+      } catch (LockException le) {
+        throw new CommitStateUnknownException(
```

Comment:
Can we extract all common code between `HiveTableOperations` and `HiveViewOperations` to a Util?

Reply:
IMO, we should do this with another patch.

Reply:
IMO, While adding only we should refactor it to avoid duplicate codes. Lets see what others think.

Reply:
@nastra: what do you think about this?

Reply:
does this also need to set the commit status, similar to https://github.com/apache/iceberg/blob/0c5456068c9126ed099c140195afefb7f950ace9/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java#L253?

Reply:
IMO refactoring should be a separate PR, otherwise reviewing will be quite difficult due to lots of changes

Reply:
@nastra , In case of failed commit, it will have an orphan metadata file. Since `view` is not heavy write operation do you think adding such checks are a good idea? 
Although for a long run `View` can have many metadata files? Do we have any plan to run GC for views too?